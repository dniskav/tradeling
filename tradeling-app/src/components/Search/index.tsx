import React, { useEffect } from 'react';
import { useTranslation } from 'react-i18next';
import DropDown from '../UI/DropDown';
import TextField from '../UI/TextField';
import styled from 'styled-components';
import { useDispatch, useSelector } from 'react-redux';
import { setSearchKind, setSearchTerm, fetchItemsList, setitemsList, setPage, clearList } from '../../redux/actions/index';
import { getKind, getTerm } from '../../redux/selectors/index';

const Container = styled.div`
  display: flex;
  margin-top: 20px;
  justify-content: space-between;
  width: 480px;
`;

const Search: React.FC = () => {
  const { t } = useTranslation();
  const dispatch = useDispatch();
  const term = useSelector(getTerm);
  const kind = useSelector(getKind);

  useEffect(() => {
    if(term.length > 2) {
      dispatch(fetchItemsList());
    } else {
      dispatch(setitemsList([]));
      dispatch(clearList());
    }
    dispatch(setPage(1));
  }, [term, kind, dispatch]);

  return(
    <Container>
      <TextField placeholder={t('Search.caption')} onChange={(e: any) => dispatch(setSearchTerm(e.target.value))} value={term}/>
      <DropDown value={kind} onChange={(e: any) => dispatch(setSearchKind(e))}>
        <option value="users">{t('Search.user')}</option>
        <option value="repositories">{t('Search.repo')}</option>
      </DropDown>
    </Container>
  )
};

export default Search;
