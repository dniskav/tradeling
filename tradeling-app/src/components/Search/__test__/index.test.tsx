import Search from "../index";
import { shallow } from "enzyme";

jest.mock("react-redux", () => {
  return {
    useSelector: () => [],
    useDispatch: () => jest.fn(),
  }
});

describe("<Results />", () => {
  const tree = shallow(<Search />);
  it("Search should match snapshot", () => {
    expect(tree).toMatchSnapshot();
  });
});
