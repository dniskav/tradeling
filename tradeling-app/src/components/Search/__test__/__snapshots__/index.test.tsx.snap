// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Results /> Search should match snapshot 1`] = `
<styled.div>
  <TextField
    onChange={[Function]}
    placeholder="Search.caption"
    value={Array []}
  />
  <DropDown
    onChange={[Function]}
    value={Array []}
  >
    <option
      value="users"
    >
      Search.user
    </option>
    <option
      value="repositories"
    >
      Search.repo
    </option>
  </DropDown>
</styled.div>
`;
